### PR TITLE
[issue-635]add handle for avg(stringType)+IntType query

### PR DESCRIPTION
**Pull request by Jay357089**
issue #635 

the stage failure is because avg(StringType) is resolved as String Type and IntType resolved as DoubleType, the type of both plus can not be resolved. 
i cast it to DoubleType.

### CODE REVIEW

**Comment by Jay357089** [comment]
@kumarvishal09  @QiangCai  pls review.

**Comment by kumarvishal09** [comment]
please handle for other aggregate function sum, max, min, sum distinct for example sum(String)+10 in this can also we need to show null

**Comment by Jay357089** [comment]
@kumarvishal09 handled.

**Comment by Jay357089** [comment]
@kumarvishal09 pls review and check in.

**Comment by Jay357089** [comment]
retest this please

**Comment by kumarvishal09** [comment]
retest this please

File: integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala [hunk 1]
```diff
@@ -111,7 +111,8 @@ case class AverageCarbon(child: Expression, castedDataType: DataType = null)
     val partialSum = Alias(AverageCarbon(child), "PartialAverage")()
     SplitEvaluation(
       AverageCarbonFinal(partialSum.toAttribute,
-        if (child.dataType == IntegerType) DoubleType else child.dataType),
+        if (child.dataType == IntegerType || child.dataType == StringType) DoubleType
```
Comment (ravipesala) [inline integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala]: what if the datatype is timestamp?It fails in this case so please handle that scenario as well.

**Comment by kumarvishal09** [comment]
@Jay357089 you can check how other aggregate functions are working you can use the same code

File: integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala (hunk 1 above)
Comment (Jay357089) [inline integration/spark/src/main/scala/org/carbondata/spark/agg/CarbonAggregates.scala]: @ravipesala  i will handle

**Comment by Jay357089** [comment]
@kumarvishal09 there is some difference between avg and other aggregation, because avg(int) should be double, and for dimension, int is handled as int, but for measure, int is handled as double. so, some specific handle should be done to avg.

**Comment by Jay357089** [comment]
@ravipesala handled, also added test case. pls review and check in.

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**